File: services/payment-service/src/main/java/io/minishop/payment/config/MockPgProperties.java
package io.minishop.payment.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "mini-shop.mock-pg")
public record MockPgProperties(
        boolean enabled,
        long latencyMeanMs,
        long latencyStddevMs,
        double failureRate,
        double timeoutRate,
        long timeoutMs
) {}

File: services/payment-service/src/main/kotlin/io/minishop/payment/config/MockPgProperties.kt
package io.minishop.payment.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "mini-shop.mock-pg")
@JvmRecord
data class MockPgProperties(
    val enabled: Boolean,
    val latencyMeanMs: Long,
    val latencyStddevMs: Long,
    val failureRate: Double,
    val timeoutRate: Double,
    val timeoutMs: Long,
)
